plugins/fileservices/fileservices.cpp

```diff
@@ -2774,14 +2817,25 @@ FILESERVICES_API void  FILESERVICES_CALL fsMoveExternalFile(ICodeContext * ctx,c
     AuditMessage(ctx,"MoveExternalFile",frompath,topath);
 }
 
+FILESERVICES_API void  FILESERVICES_CALL fsMoveExternalFile(ICodeContext *ctx,const char *location,const char *frompath,const char *topath)
+{
+    implementMoveExternalFile(ctx,location,frompath,topath,nullptr);
+}
+
+FILESERVICES_API void  FILESERVICES_CALL fsMoveExternalFile_v2(ICodeContext *ctx,const char *location,
+    const char *frompath,const char *topath,const char *planename)
+{
+    implementMoveExternalFile(ctx,location,frompath,topath,planename);
+}
+
 FILESERVICES_API void  FILESERVICES_CALL fsDeleteExternalFile(ICodeContext * ctx,const char *location,const char *path)
 {
     SocketEndpoint ep(location);
     if (ep.isNull())
         throw MakeStringException(-1,"fsDeleteExternalFile: Cannot resolve location %s",location);
     CDfsLogicalFileName lfn;
     lfn.setExternal(location,path);
-    checkExternalFileRights(ctx,lfn,false,true);
+    checkExternalFileRights(ctx,lfn,false,false,true); //The path could be a directory. So, it may need to be checked as a scope. Will re-visit this when adding plane name here.
```

Comment:
agree. CreateExternalDirectory also implies it must be a directory.
I think all cases need to ensure that the lfn passed to checking function is a directory.